# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLMediaElement.cpp
# Commit: d0e8d413cd1c
# Full Hash: d0e8d413cd1c6aa414cf167b085f635ae857e408
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-10-02 16:35:54
# Regressor Bug: 1493613
# File Overlap Count: 1
# Description:
#   Bug 1454998 - Remove track blocking from MediaInputPort. r=karlt
#   
#   This functionality is not used since we moved to only having a single track per
#   MediaStream (bug 1493613).
#   
# ==============================================================================

diff -r 3ce4dc7e9ae2 -r d0e8d413cd1c dom/html/HTMLMediaElement.cpp
--- a/dom/html/HTMLMediaElement.cpp	Wed Oct 02 08:12:18 2019 +0000
+++ b/dom/html/HTMLMediaElement.cpp	Wed Oct 02 08:12:28 2019 +0000
@@ -657,15 +657,18 @@
                                            MediaStreamTrackSource)
 
   StreamCaptureTrackSource(MediaStreamTrackSource* aCapturedTrackSource,
-                           ProcessedMediaStream* aStream, MediaInputPort* aPort)
+                           ProcessedMediaStream* aStream, MediaInputPort* aPort,
+                           TrackID aTrackID)
       : MediaStreamTrackSource(aCapturedTrackSource->GetPrincipal(),
                                nsString()),
         mCapturedTrackSource(aCapturedTrackSource),
         mStream(aStream),
-        mPort(aPort) {
+        mPort(aPort),
+        mTrackID(aTrackID) {
     MOZ_ASSERT(mCapturedTrackSource);
     MOZ_ASSERT(mStream);
     MOZ_ASSERT(mPort);
+    MOZ_ASSERT(IsTrackIDExplicit(mTrackID));
 
     mCapturedTrackSource->RegisterSink(this);
   }
@@ -674,9 +677,9 @@
     if (!mStream) {
       return;
     }
-    mStream->SetTrackEnabled(mPort->GetDestinationTrackId(),
-                             aEnabled ? DisabledTrackMode::ENABLED
-                                      : DisabledTrackMode::SILENCE_FREEZE);
+    mStream->SetTrackEnabled(mTrackID, aEnabled
+                                           ? DisabledTrackMode::ENABLED
+                                           : DisabledTrackMode::SILENCE_FREEZE);
   }
 
   void Destroy() override {
@@ -754,6 +757,7 @@
   RefPtr<MediaStreamTrackSource> mCapturedTrackSource;
   RefPtr<ProcessedMediaStream> mStream;
   RefPtr<MediaInputPort> mPort;
+  const TrackID mTrackID;
 };
 
 NS_IMPL_ADDREF_INHERITED(HTMLMediaElement::StreamCaptureTrackSource,
@@ -3195,8 +3199,8 @@
 
   ProcessedMediaStream* stream = inputTrack->Graph()->CreateTrackUnionStream();
   RefPtr<MediaInputPort> port = inputTrack->ForwardTrackContentsTo(stream);
-  auto source = MakeRefPtr<StreamCaptureTrackSource>(&inputTrack->GetSource(),
-                                                     stream, port);
+  auto source = MakeRefPtr<StreamCaptureTrackSource>(
+      &inputTrack->GetSource(), stream, port, inputTrack->GetTrackID());
 
   // Track is muted initially, so we don't leak data if it's added while paused
   // and an MSG iteration passes before the mute comes into effect.